File: server/model/GameListModel.java
package model;

import java.util.*;

public class GameListModel
{
    private List<LobbyGameModel> games;
    private Deck destDeck;
    private Deck trainDeck;

    public GameListModel()
    {
        games = new ArrayList<>();
    }

    public void addGame(LobbyGameModel game)
    {
        assert(game instanceof LobbyGameModel);
        assert(game != null);
        games.add(game);
    }
    public void removeGame(LobbyGameModel game)
    {
        assert(game instanceof LobbyGameModel);
        assert(game != null);
        games.remove(game);
    }
    public void removeGame(String gameID)
    {
        assert(gameID != null);
        assert(!gameID.isEmpty());

        for(int i = 0; i < games.size(); i++)
        {
            if(games.get(i).getGameID().equals(gameID))
                games.remove(i);
        }
    }
    public LobbyGameModel getGameByID(String gameID)
    {
        assert(gameID != null);
        assert(!gameID.isEmpty());
        for(int i = 0; i < games.size(); i++)
        {
            if(games.get(i).getGameID().equals(gameID))
                return games.get(i);
        }
        return null;
    }
    public List<LobbyGameModel> getGameList()
    {
        return games;
    }


}

File: client/src/Models/GameListModel.ts
import {LobbyGame} from "./LobbyGameModel"
export class GameList {
    games: Array<LobbyGame>;

    constructor() {
        this.games = new Array<LobbyGame>();
    }

    addGame(game: LobbyGame): void {
        this.games.push(game);
    }

    removeGame(gameId: string): void {
        for (var i = this.games.length - 1; i >= 0; --i) {
            if (this.games[i].getGameID() == gameId) {
                this.games.splice(i, 1);
            }
        }
    }

    replaceGameList(newGameList: Array<LobbyGame>): void {
        this.games = newGameList;
    }

    getGames(): Array<LobbyGame> {
        return this.games;
    }

}
